tools/range_read/range_read.c

```diff
@@ -305,49 +327,40 @@ cursor(
                       be64toh(*(uint64_t *)key),
                       be64toh(*((uint64_t *)key + 1)));
         }
+        t_read = get_time_ns();
+
+        if (!opts.use_update)
+            kh_cursor_destroy(c);
 
-        kh_cursor_destroy(c);
+        t_full = get_time_ns();
 
-        if ((opcnt % dt->dt_skip == 0) && dt_idx < dt_size)
-            dt->dt[dt_idx++] = get_time_ns() - start;
+        hdr_record_value(lat->lat_create, t_create - t_start);
+        hdr_record_value(lat->lat_seek, t_seek - t_create);
+        hdr_record_value(lat->lat_read, (t_read - t_seek) / (i ?: 1));
+        hdr_record_value(lat->lat_full, t_full - t_start);
```

Comment:
Would be curious to know what the hit to the throughput this imposes...

Reply:
Hmm, I though I'd be avoiding the atomics since this uses hdr_record_value() and not hdr_record_value_atomic()...

Reply:
i could be wrong, but looks like that ends up calling count_inc_normalised() which does two atomic-fetch-adds...   or at least purports to, it's all abstracted, dunno what's really happening...

Reply:
you could do a run with those calls ifdef'd out and see if it makes any difference in throughput, if you were so inclined :-)  i would think it would be a very small impact unless there's contention somewhere...
Ewwww, this was clearly written for windoze.  If windoze isn't the target then they use assembly if the target is amd64 (except for atomic-add-fetch they call __sync_add_and_fetch(), and otherwise punt.   What a mess..

Reply:
Crap, they don't use atomics in your use case, my bad, you were right Gaurav!